### PR TITLE
feat: add privilegedhostexec collector

### PR DESCRIPTION
**Pull request by mhrabovcin**
JIRA: https://jira.d2iq.com/browse/D2IQ-79179

### CODE REVIEW

**Review by jongiddy — APPROVED** [review]

File: pkg/collect/privileged_host_collector.go [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+// container that collects data as an `initContainer` and it expects that the
+// container runs one-off process with an expectation of a clean exit. Once the
+// initContainer is completed an additional `pause` container is launched that
+// is a no-op container. This container only waits and allows the main
+// support-bundle process to collect produced data (copy from the container).
+// The `initContainer` and `pause` containers share a volume to which the
+// container producing the support bundle data writes. Data written to other
+// paths by the `initContainer` will not be copied back to the support bundle.
+
+// This collector is a combination of `CopyFromHost` and `Exec` containers.
```
Comment (tillt) [inline pkg/collect/privileged_host_collector.go:39]: Great comment - very helpful.

**Review by tillt — APPROVED** [review]
We talked about testing but not sure what the outcome for this particular collector was. In any case, looks great and works -> superbe.

File: pkg/apis/troubleshoot/v1beta2/collector_shared.go [hunk 2]
```diff
@@ -114,6 +114,36 @@ type CopyFromHost struct {
 	ExtractArchive  bool              `json:"extractArchive,omitempty" yaml:"extractArchive,omitempty"`
 }
 
+type PrivilegedHostExec struct {
```
[the remark this answers is not in the document]
Comment (mhrabovcin) [reply pkg/apis/troubleshoot/v1beta2/collector_shared.go]: I've renamed the collector to `ExecCopyFromHost`.

**Comment by mhrabovcin** [comment]
@tillt I would like to add tests to `dkp-diagnostics` repo.

**Pull request merged**